### ISSUE TITLE
End to end test to cover line number for IL2CPP

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-unity#900]
We need a test that validates our IL2CPP line number support doesn't regress.
This is particularly important when we bump Unity in CI. And it might be too slow or flaky to run in normal PRs (that are unlikely to break this).

The goal is to read the event from Sentry in the tests, and make sure it contains line numbers.

The React Native SDK repository has an e2e test, the API calls can be seeing there:

* https://github.com/getsentry/sentry-react-native/blob/cd190906d8962d4896251385bf7e4853ef9ccbc1/sample/test/e2e.test.ts
* https://github.com/getsentry/sentry-react-native/blob/cd190906d8962d4896251385bf7e4853ef9ccbc1/.github/workflows/e2e.yml

**Comment by vaind** [comment]
I already mentioned this as a TODO in the other IL2CPP line number support issue: #895 - I thought we could just reuse SmokeTester.cs? Or are the line numbers only assigned on the server?

**Comment by bitsandfoxes** [comment]
Yes, the symbolication of the native stacktrace back to C# line numbers happens on the server.
